src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt

```diff
@@ -478,6 +480,14 @@ private fun checkPatRange(holder: AnnotationHolder, element: RsPatRange) {
 
 private fun checkTraitType(holder: AnnotationHolder, element: RsTraitType) {
     if (element.impl != null) return
+    if (element.polyboundList.count { it.bound.lifetime != null } > 1) {
+        val fixes = mutableListOf<LocalQuickFix>()
+        for (e in element.polyboundList.filter { it.bound.lifetime != null }) {
+            fixes += RemovePolyBoundFix(e)
+        }
+        RsDiagnostic.TooManyLifetimeBoundsOnTraitObjectError(element, fixes).addToHolder(holder)
+    }
+
```

Comment:
There are many repeating operation on the PSI tree and it seems trivial to avoid them (at least extract `element.polyboundList` to a variable, but better use a single loop for counting bounds with/without lifetimes)